pkg/controller/kieapp/defaults/auth.go

```diff
@@ -0,0 +1,87 @@
+package defaults
+
+import (
+	"github.com/kiegroup/kie-cloud-operator/pkg/apis/app/v1"
+	"github.com/kiegroup/kie-cloud-operator/pkg/controller/kieapp/constants"
+	"github.com/kiegroup/kie-cloud-operator/pkg/controller/kieapp/shared"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const ssoHostnameVar = "HOSTNAME_HTTPS"
+const ssoClientVar = "SSO_CLIENT"
+
+// ConfigureHostname sets the HOSTNAME_HTTPS environment variable with the provided hostname
+// IF not yet set AND SSO auth is configured AND SSO_CLIENT exists
+func ConfigureHostname(object *v1.CustomObject, cr *v1.KieApp, hostname string) {
+	if cr.Spec.Auth.SSO == nil {
+		return
+	}
+	for dcIdx := range object.DeploymentConfigs {
+		dc := &object.DeploymentConfigs[dcIdx]
+		for containerIdx := range dc.Spec.Template.Spec.Containers {
+			container := &dc.Spec.Template.Spec.Containers[containerIdx]
+			if pos := shared.GetEnvVar(ssoClientVar, container.Env); pos == -1 {
+				continue
+			}
+			if pos := shared.GetEnvVar(ssoHostnameVar, container.Env); pos == -1 {
+				container.Env = append(container.Env, corev1.EnvVar{
+					Name:  ssoHostnameVar,
+					Value: hostname,
+				})
+			} else if len(container.Env[pos].Value) == 0 {
+				container.Env[pos].Value = hostname
+			}
+		}
+	}
+}
+
+func configureAuth(spec v1.KieAppSpec, envTemplate *v1.EnvTemplate) (err error) {
```

Comment:
When you name your return variable, you no longer have to return it in the function. You just need to set it to a value and do a plain return. Ideally and in the case of if/elseif, that would be:
```
if() {
err = ...
} else if() {
err = ...
} else {
err = ...
}
return
```

If that no longer makes sense because of how conditions changed, maybe best to just remove the named return variable because it's not being used in this function today.